src/openms/source/ANALYSIS/OPENSWATH/DIAPrescoring.cpp

```diff
@@ -130,59 +130,67 @@ namespace OpenMS
     std::transform(intExp.begin(), intExp.end(), intExp.begin(), OpenSwath::mySqrt());
-    std::transform(theorint.begin(), theorint.end(), theorint.begin(), OpenSwath::mySqrt());
-
-    double intExptotal = std::accumulate(intExp.begin(), intExp.end(), 0.0);
-    double intTheorTotal = std::accumulate(theorint.begin(), theorint.end(), 0.0);
-
-    OpenSwath::normalize(intExp, intExptotal, intExp);
-    OpenSwath::normalize(theorint, intTheorTotal, theorint);
+    std::transform(intTheor.begin(), intTheor.end(), intTheor.begin(), OpenSwath::mySqrt());
 
-    manhattan = OpenSwath::manhattanDist(intExp.begin(), intExp.end(), theorint.begin());
+    double intExpTotal = std::accumulate(intExp.begin(), intExp.end(), 0.0);
+    double intTheorTotal = std::accumulate(intTheor.begin(), intTheor.end(), 0.0);
 
-    //std::cout << spectrum.size() << std::endl;
-    DIAHelpers::addPreisotopeWeights(firstIstotope, spectrum2, 2, -0.5);
-    std::vector<double>  theorint2;
-    DIAHelpers::extractSecond(spectrum, theorint2);
-    std::transform(theorint2.begin(), theorint2.end(), theorint2.begin(), OpenSwath::mySqrt());
+    OpenSwath::normalize(intExp, intExpTotal, intExp);
+    OpenSwath::normalize(intTheor, intTheorTotal, intTheor);
 
-    intExptotal = OpenSwath::norm(intExp.begin(), intExp.end());
-    intTheorTotal = OpenSwath::norm(theorint2.begin(), theorint2.end());
+    manhattan = OpenSwath::manhattanDist(intExp.begin(), intExp.end(), intTheor.begin());
 
-    OpenSwath::normalize(intExp, intExptotal, intExp);
-    OpenSwath::normalize(theorint2, intTheorTotal, theorint2);
+    // compare against the spectrum with negative weight preIsotope peaks
+    std::vector<double> intTheor2;
+    // WARNING: This was spectrumWIso and therefore with 0 preIso weights before! Was this a bug?
+    // Otherwise we dont need the second spectrum at all.
+    DIAHelpers::extractSecond(spectrumWIsoNegPreIso, intTheor2);
+    intTheorTotal = OpenSwath::norm(intTheor2.begin(), intTheor2.end()); // use norm since we want absolute values
+    OpenSwath::normalize(intTheor2, intTheorTotal, intTheor2);
+    // Sqrt does not work if we actually have negative values now
+    //std::transform(intTheor2.begin(), intTheor2.end(), intTheor2.begin(), OpenSwath::mySqrt());
 
-    //    std::copy(intExp.begin(), intExp.end(), std::ostream_iterator<double>(std::cout, ", "));
-    //    std::cout << std::endl;
-    //    std::copy(theorint2.begin(), theorint2.end(), std::ostream_iterator<double>(std::cout, ", "));
-    //    std::cout << std::endl;
-    dotprod = OpenSwath::dotProd(intExp.begin(), intExp.end(), theorint2.begin());
+    dotprod = OpenSwath::dotProd(intExp.begin(), intExp.end(), intTheor2.begin());
   }
 
   void DiaPrescore::updateMembers_()
   {
     dia_extract_window_ = (double) param_.getValue(
       "dia_extraction_window");
     nr_isotopes_ = (int) param_.getValue("nr_isotopes");
+    //TODO nr_charges_ is never used???
```

Comment:
Hm, seem to be not actively used.
That is the only occurrence as far as I have seen:
DIAHelpers::addIsotopes2Spec(res, spectrum, nr_charges_);
https://github.com/OpenMS/OpenMS/blob/develop/src/openms/source/ANALYSIS/OPENSWATH/DIAPrescoring.cpp#L140

Reply:
Yes which was wrong in my opinion and was changed here. It was used instead of nr isotopes. What do you think?